pkg/status/statuscollector.go

```diff
@@ -41,7 +44,19 @@ func (c *Controller) syncStatusCollector(ctx context.Context, ref string) error
 		statusCollector.Name = ref
 	}
 
-	// TODO: validate the statusCollector if not nil
+	// Validate the StatusCollector
+	if errs := c.validateStatusCollector(statusCollector); len(errs) > 0 {
+		if err := c.updateStatusCollectorErrors(ctx, statusCollector, errs); err != nil {
+			return err
+		}
+
+		isDeleted = true // invalid statuscollector, should be deleted if it exists
```

Comment:
I presume you mean that this method should pretend that the StatusCollector does not exist, rather than actively delete it. If so then more clarity in the comment would be good.